prow/crier/reporters/gerrit/reporter.go

```diff
@@ -118,10 +124,16 @@ func NewReporter(cfg config.Getter, cookiefilePath string, projects map[string][
 	// line arg(which is going to be deprecated).
 	gc.Authenticate(cookiefilePath, "")
 
-	return &Client{
+	c := &Client{
 		gc:     gc,
 		lister: lister,
-	}, nil
+		prLocks: criercommonlib.NewShardedLock(
+			semaphore.NewWeighted(1),
+			map[criercommonlib.SimplePull]*semaphore.Weighted{}),
+	}
+
+	c.prLocks.RunCleanup()
```

Comment:
Why is this line needed?

Reply:
This is an async process keeps on cleaning up locks, it's useful to avoid accumulated memory allocation in a long running pod

Reply:
Sorry, I missed that this is done for the GitHub reporter already (we're just mimicking that here). LGTM